x-pack/filebeat/input/netflow/integration_test.go

```diff
@@ -0,0 +1,396 @@
+		require.NoError(t, err)
+		return eventsCount >= totalPackets
+	}, waitFor, tick)
+}
+
+type unitPayload map[string]interface{}
+
+func (u unitPayload) streamStatusEquals(streamID string, expected map[string]interface{}) bool {
+	if u == nil {
+		return false
+	}
+
+	streams, ok := u["streams"].(map[string]interface{})
+	if !ok || streams == nil {
+		return false
+	}
+
+	streamMap, ok := streams[streamID].(map[string]interface{})
+	if !ok || streamMap == nil {
+		return false
+	}
+
+	return reflect.DeepEqual(streamMap, expected)
+}
+
+func extractStateAndPayload(observed *proto.CheckinObserved, inputID string) (proto.State, unitPayload) {
+	for _, unit := range observed.GetUnits() {
+		if unit.Id == inputID {
+			return unit.GetState(), unit.Payload.AsMap()
+		}
+	}
+
+	return -1, nil
+}
+
+type DataStream struct {
+	Name   string `json:"name"`
+	Status string `json:"status"`
+}
+
+type DataStreamResult struct {
+	DataStreams []DataStream `json:"data_streams"`
+	Error       interface{}  `json:"error"`
+}
+
+func HasDataStream(ctx context.Context, username string, password string, url string, name string) error {
+	resultBytes, err := request(ctx, http.MethodGet, username, password, fmt.Sprintf("%s/_data_stream/%s", url, name))
+	if err != nil {
+		return err
+	}
+
+	if resultBytes == nil {
+		return errors.New("http not found error")
+	}
+
+	var results DataStreamResult
+	err = json.Unmarshal(resultBytes, &results)
+	if err != nil {
+		return err
+	}
+
+	if results.Error != nil {
+		return fmt.Errorf("error %v while checking for data stream %s", results.Error, name)
+	}
+
+	if len(results.DataStreams) != 1 {
+		return fmt.Errorf(
+			"unexpected count %v of data streams returned when looking for %s",
+			len(results.DataStreams), name)
+	}
+
+	if results.DataStreams[0].Name != name {
+		return fmt.Errorf("unexpected data stream %s returned when looking for %s",
+			results.DataStreams[0].Name,
+			name)
+	}
+
+	return nil
+}
+
+// Hit represents a single search hit.
+type Hit struct {
+	Index  string                 `json:"_index"`
+	Type   string                 `json:"_type"`
+	ID     string                 `json:"_id"`
+	Score  float64                `json:"_score"`
+	Source map[string]interface{} `json:"_source"`
+}
+
+// Hits are the collections of search hits.
+type Hits struct {
+	Total json.RawMessage // model when needed
+	Hits  []Hit           `json:"hits"`
+}
+
+// SearchResults are the results returned from a _search.
+type SearchResults struct {
+	Took   int
+	Hits   Hits                       `json:"hits"`
+	Shards json.RawMessage            // model when needed
+	Aggs   map[string]json.RawMessage // model when needed
+}
+
+func DataStreamEventsCount(ctx context.Context, username string, password string, url string, name string) (int, error) {
+	resultBytes, err := request(ctx, http.MethodGet, username, password, fmt.Sprintf("%s/%s/_search?q=!error.message:*", url, name))
```

Comment:
Since this is filtering out errors then I think it should consider [malformed](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-ignored-field.html) data as an error too.

`_search?q=!_ignored:* AND !event.message:*`

Reply:
nice catch thanks!